Uses PHP8's constructor property promotion in core/Command and /

fsamapoor opened the pull request:
Following https://github.com/nextcloud/server/pull/38636, https://github.com/nextcloud/server/pull/38637, and https://github.com/nextcloud/server/pull/38638 PRs, I have made the required adjustments to the `/core/Command/` namespace as well.

I figured I should split the changes into multiple PRs to make reviewing the changes easier.

This PR refactors `/core/Command` and `/core/Command/TwoFactorAuth` classes by using PHP8's constructor property promotion to remove redundant lines and make the code cleaner.

Commit:
Uses PHP8's constructor property promotion.
 in core/Command and /TwoFactorAuth classes.

Signed-off-by: Faraz Samapoor <fsa@adlas.at>

come-nc commented on core/Command/TwoFactorAuth/Disable.php:
```diff
@@ -29,12 +29,11 @@
-	public function __construct(ProviderManager $manager, IUserManager $userManager) {
+	public function __construct(
+		private ProviderManager $manager,
+		protected IUserManager $userManager,
```
This is defined in Base class already.
The way this is currently done is really ugly, so a refactor is needed.
A constructor needs to be added in Base which takes `$userManager` as parameter.

artonge replied:
@fsamapoor can you address the comment?

fsamapoor replied:
@come-nc Sorry about the delayed changes. I hope I did it right.

Commit:
Adds constructor to the Based class.

Based on:
https://github.com/nextcloud/server/pull/38775#discussion_r1227641788

Signed-off-by: Faraz Samapoor <fsa@adlas.at>

come-nc commented on core/Command/TwoFactorAuth/Cleanup.php:
```diff
@@ -27,17 +27,20 @@
-		$this->registry = $registry;
+	public function __construct(
+		private IRegistry $registry,
+		protected IUserManager $userManager,
```
```suggestion
		IUserManager $userManager,
```

Fenn-CS replied:
Would this not reduce them to ordinary parameters?

fsamapoor replied:
It would. The thing is it is being passed to the constructor of the parent (Base) class and would be available via inheritance in the child classes. as well

Fenn-CS replied:
Right, thanks!

come-nc commented on core/Command/TwoFactorAuth/Disable.php:
```diff
@@ -29,12 +29,14 @@
-		$this->userManager = $userManager;
+	public function __construct(
+		private ProviderManager $manager,
+		protected IUserManager $userManager,
```
```suggestion
		IUserManager $userManager,
```

come-nc commented on core/Command/TwoFactorAuth/Enable.php:
```diff
@@ -29,12 +29,14 @@
-		$this->userManager = $userManager;
+	public function __construct(
+		private ProviderManager $manager,
+		protected IUserManager $userManager,
```
```suggestion
		IUserManager $userManager,
```

come-nc commented on core/Command/TwoFactorAuth/Enforce.php:
```diff
@@ -35,12 +35,8 @@
-	private MandatoryTwoFactor $mandatoryTwoFactor;
-
-	public function __construct(MandatoryTwoFactor $mandatoryTwoFactor) {
+	public function __construct(private MandatoryTwoFactor $mandatoryTwoFactor) {
```
```suggestion
	public function __construct(
		private MandatoryTwoFactor $mandatoryTwoFactor,
	) {
```

come-nc commented on core/Command/Check.php:
```diff
@@ -29,11 +29,8 @@
-	private SystemConfig $config;
-
-	public function __construct(SystemConfig $config) {
+	public function __construct(private SystemConfig $config) {
```
```suggestion
	public function __construct(
		private SystemConfig $config,
	) {
```

come-nc commented on core/Command/TwoFactorAuth/State.php:
```diff
@@ -33,13 +33,14 @@
-		$this->userManager = $userManager;
+	public function __construct(
+		private IRegistry $registry,
+		protected IUserManager $userManager,
```
```suggestion
		IUserManager $userManager,
```

nfebe reviewed (approved):
Thank you!

fsamapoor committed:
Update core/Command/Check.php

Co-authored-by: Côme Chilliet <91878298+come-nc@users.noreply.github.com>
Signed-off-by: Faraz Samapoor <f.samapoor@gmail.com>

fsamapoor committed:
Update core/Command/TwoFactorAuth/Enforce.php

Co-authored-by: Côme Chilliet <91878298+come-nc@users.noreply.github.com>
Signed-off-by: Faraz Samapoor <f.samapoor@gmail.com>

fsamapoor committed:
Update core/Command/TwoFactorAuth/State.php

Co-authored-by: Côme Chilliet <91878298+come-nc@users.noreply.github.com>
Signed-off-by: Faraz Samapoor <f.samapoor@gmail.com>

fsamapoor committed:
Update core/Command/TwoFactorAuth/Disable.php

Co-authored-by: Côme Chilliet <91878298+come-nc@users.noreply.github.com>
Signed-off-by: Faraz Samapoor <f.samapoor@gmail.com>

fsamapoor committed:
Update core/Command/TwoFactorAuth/Enable.php

Co-authored-by: Côme Chilliet <91878298+come-nc@users.noreply.github.com>
Signed-off-by: Faraz Samapoor <f.samapoor@gmail.com>

fsamapoor committed:
Update core/Command/TwoFactorAuth/Cleanup.php

Co-authored-by: Côme Chilliet <91878298+come-nc@users.noreply.github.com>
Signed-off-by: Faraz Samapoor <f.samapoor@gmail.com>

artonge reviewed (changes requested):
> There was 1 error:
>
> 1) Core\Command\TwoFactorAuth\CleanupTest::testCleanup
> ArgumentCountError: Too few arguments to function OC\Core\Command\TwoFactorAuth\Cleanup::__construct(), 1 passed in  /drone/src/tests/Core/Command/TwoFactorAuth/CleanupTest.php on line 47  and exactly 2 expected
>
> /drone/src/core/Command/TwoFactorAuth/Cleanup.php:36
> /drone/src/tests/Core/Command/TwoFactorAuth/CleanupTest.php:47

Commit:
Fixes testcase error.

Signed-off-by: Faraz Samapoor <fsa@adlas.at>

artonge commented:
@come-nc
